pymodbus/server/simulator/http_server.py

```diff
@@ -260,7 +260,6 @@ async def stop(self):
         self.site = None
         if not self.serving.done():
             self.serving.set_result(True)
-        await asyncio.sleep(0)
```

Comment:
As I just wrote on the issue, this used to be needed to allow other asyncio task to run (in this case the effect of the future set above).

A larger number of the old sporadic test errors was due to the fact that other tasks did not get a chance to run, and I am pretty sure this is the same here.

Reply:
Sure, but look in context - it's the very last line of the function, so it would yield to other tasks anyways.

The only possible us would be hoping some other task runs before `await close()` returns...
(1) I don't see any calling code that would need this
(2) It's a very strange pattern anyways - if `close()` needs some other task to run it should just `await` that task.

Reply:
At least 3.8 do not guarantee a task switch at return, and I think it is a pattern that are kept in newer versions as well.

It is easy for you to say that close() should "just" await....but often it does not know the task, like in this case, who is waiting on the future.

Remember, when a function calls "await stop()" it expect everything to have been cleaned up, without the sleep(0) the future might still be dangling (this is not theoretical I hunted these bugs, when cleaning up sporadic test errors).

Reply:
OK, in that case I will close this PR (and the issue).